### ISSUE TITLE
No unused expressions fails with pipeline operator

### ISSUE THREAD

**Issue by jhpratt** [babel/eslint-plugin-babel#174]
I have the following code which triggers the lint, but shouldn't.

```js
sections_elem.requestUpdate() |> promises.push
```

I _believe_ it's referring to the `promises.push`. It _is_ used, as it's implicitly called with the pipeline operator.

**Comment by ljharb** [comment]
The result of the pipeline isn't used, though?

**Comment by jhpratt** [comment]
Not entirely sure what you mean. The result is being pushed onto the array, which is _later_ `await`ed. This is no different than calling `promises.push(sections
_elem.requestUpdate())`, aside from the order of execution (which is irrelevant here). As far as I'm aware, this is a perfectly valid use under the proposal.

**Comment by mikestopcontinues** [comment]
Here's another example, from React:

```js
useEffect(() => {
    getAllNodes()
    |> R.filter(({meta: {type}}) => type === 'post', #)
    |> R.sortWith([R.descend(R.path(['meta', 'date']))], #)
    |> setState;
  }, []);
```

I think the logic should be: if the end of a pipeline is a function call, then the value is used.

**Comment by MikeMcElroy** [comment]
When we evaluated the pipeline operator for use on our team, we scrapped the idea because of this broken rule. It'd be awesome to see the PR above get finessed and accepted so we can start using it.

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
